netezza-plugin/src/main/java/co/cask/netezza/NetezzaDBRecord.java

```diff
@@ -53,25 +53,22 @@ public NetezzaDBRecord() {}
   protected void handleField(ResultSet resultSet, StructuredRecord.Builder recordBuilder, Schema.Field field,
                              int sqlType, int sqlPrecision, int sqlScale) throws SQLException {
     if (netezzaTypes.contains(sqlType)) {
-      handleOracleSpecificType(resultSet, recordBuilder, field, sqlType);
+      handleNetezzaSpecificType(resultSet, recordBuilder, field, sqlType);
     } else {
       setField(resultSet, recordBuilder, field, sqlType, sqlPrecision, sqlScale);
     }
   }
 
-  private void handleOracleSpecificType(ResultSet resultSet,
-                                        StructuredRecord.Builder recordBuilder, Schema.Field field,
-                                        int sqlType) throws SQLException {
-
-    Object original = resultSet.getObject(field.getName());
+  private void handleNetezzaSpecificType(ResultSet resultSet,
+                                         StructuredRecord.Builder recordBuilder, Schema.Field field,
+                                         int sqlType) throws SQLException {
 
     switch (sqlType) {
       case Types.VARBINARY:
-        original = resultSet.getBytes(field.getName());
-        recordBuilder.set(field.getName(), original);
+        recordBuilder.set(field.getName(), resultSet.getBytes(field.getName()));
         break;
       case INTERVAL:
-        recordBuilder.set(field.getName(), original.toString());
+        recordBuilder.set(field.getName(), resultSet.getString(field.getName()));
```

Comment:
how come this changed from the column to the whole row? My previous question was just about what object was being used here to make sure .toString() is meaningful

Reply:
Yeah it's meaningful, it prints interval in 'n years n mons n days hh:mm:ss' format  , also INTERVAL type mapping are covered in integration tests